src/components/mgt-teams-channel-picker/mgt-teams-channel-picker.ts

```diff
@@ -0,0 +1,781 @@
+@customElement('mgt-teams-channel-picker')
+export class MgtTeamsChannelPicker extends MgtTemplatedComponent {
+  /**
+   * Array of styles to apply to the element. The styles should be defined
+   * user the `css` tag function.
+   */
+  static get styles() {
+    return styles;
+  }
+
+  /**
+   * Gets Selected item to be used
+   *
+   * @readonly
+   * @type {SelectedChannel}
+   * @memberof MgtTeamsChannelPicker
+   */
+  public get selectedItem(): SelectedChannel {
+    if (this._selectedItemState) {
+      return { channel: this._selectedItemState.item, team: this._selectedItemState.parent.item };
+    } else {
+      return null;
+    }
+  }
+
+  private set items(value) {
+    if (this._items === value) {
+      return;
+    }
+    this._items = value;
+    this._treeViewState = value ? this.generateTreeViewState(value) : [];
+    this.resetFocusState();
+  }
+  private get items(): DropdownItem[] {
+    return this._items;
+  }
+
+  // User input in search
+  @query('.team-chosen-input') private _input: HTMLInputElement;
+  private _inputValue: string = '';
+
+  private _isFocused = false;
+
+  private _selectedItemState: ChannelPickerItemState;
+  private _items: DropdownItem[];
+  private _treeViewState: ChannelPickerItemState[] = [];
+
+  // focus state
+  private _focusList: ChannelPickerItemState[] = [];
+  private _focusedIndex: number = -1;
+  private debouncedSearch;
+
+  // determines loading state
+  @property({ attribute: false }) private _isDropdownVisible;
+
+  constructor() {
+    super();
+    this.handleWindowClick = this.handleWindowClick.bind(this);
+    this.addEventListener('keydown', e => this.onUserKeyDown(e));
+    this.addEventListener('focus', _ => this.loadTeamsIfNotLoaded());
+    this.addEventListener('mouseover', _ => this.loadTeamsIfNotLoaded());
+  }
+
+  /**
+   * Invoked each time the custom element is appended into a document-connected element
+   *
+   * @memberof MgtTeamsChannelPicker
+   */
+  public connectedCallback() {
+    super.connectedCallback();
+    window.addEventListener('click', this.handleWindowClick);
+  }
+
+  /**
+   * Invoked each time the custom element is disconnected from the document's DOM
+   *
+   * @memberof MgtTeamsChannelPicker
+   */
+  public disconnectedCallback() {
+    window.removeEventListener('click', this.handleWindowClick);
+    super.disconnectedCallback();
+  }
+
+  /**
+   * selects a channel by looking up the id in the Graph
+   *
+   * @param {string} channelId MicrosoftGraph.Channel.id
+   * @returns {Promise<return>} A promise that will resolve to true if channel was selected
+   * @memberof MgtTeamsChannelPicker
+   */
+  public async selectChannelById(channelId: string): Promise<boolean> {
+    const provider = Providers.globalProvider;
+    if (provider && provider.state === ProviderState.SignedIn) {
+      // since the component normally handles loading on hover, forces the load for items
+      if (!this.items) {
+        this.requestStateUpdate();
```

Comment:
The issue of the selectChannelById not working the first time is here. This should be:

```suggestion
        await this.requestStateUpdate();
```

Currently, you are not waiting for the tree to populate so the lines below will have an empty tree.

Reply:
I see what you mean now, I wasn't quick enough to see the functionality. Updated here:

242783d